babi.py

```diff
@@ -914,7 +914,82 @@ def _get_char(stdscr: 'curses._CursesWindow') -> Key:
     return Key(wch, key, keyname)
 
 
-EditResult = enum.Enum('EditResult', 'EXIT NEXT PREV')
+EditResult = enum.Enum('EditResult', 'EXIT NEXT PREV EDIT')
+
+
+class Command:
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        keybinds: List[str],
+        func: Callable[[Screen, Key], EditResult],
+        aliases: List[str] = [],
+    ) -> None:
+        self.name = name
+        self.description = description
+        self.keybinds = keybinds
+        self.aliases = aliases
+        self.func = func
+
+    def translate_keybinds(self) -> List[bytes]:
+        """
+        Translates the keybinds in the form "Modifier-key" to curses-compatible
+        format.
+        """
+        new_binds = []  # type: List[bytes]
+        for bind in self.keybinds:
+            new_binds.append(bind.replace('C-', '^').upper().encode('UTF-8'))
+        return new_binds
+
+
+COMMANDS = {}  # type: Dict[str, Command]
+
+
+def command(
+    name: str,
+    description: str,
+    keybinds: List[str],
+    aliases: List[str] = [],
+) -> Callable[[Callable[[Screen, Key], EditResult]], None]:
+    def inner(func: Callable[[Screen, Key], EditResult]) -> None:
```

Comment:
this should return the function that got passed in, otherwise it's not really a decorator and only being used for side-effects

An alternative to using a decorator would be something like:

```python
commands.register(func, ...)
```
that way it's not purely depending on side-effects

Reply:
So there'd be a command registry of sorts? I suppose that *would* make indexing by different things easier (like aliases and keybinds)

Reply:
I managed to do this a different way.